Automattic/wp-calypso
# Fixes test/e2e dependencies

### Background

`@yarnpkg/doctor` found a few problems with the dependencies:

```
➤ YN0000: ┌ /Users/sergio/src/automattic/wp-calypso/test/e2e/package.json
...
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/test/e2e/lib/xmpp-client.js:13:17: Undeclared dependency on @xmpp/plugins
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/test/e2e/lib/xmpp-client.js:14:17: Undeclared dependency on @xmpp/plugins
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/test/e2e/package.json:55:22: Unmet transitive peer dependency on request@^2.34, via request-promise@^4.2.2
➤ YN0000: │ Resolving testarmada-magellan-mocha-plugin@github:Automattic/magellan-mocha-plugin#b731cdf errored with Couldn't resolve "b731cdf" as either a commit, a tag, or a head - if a commit, use the 40-characters commit hash
```

### Changes proposed in this Pull Request

* Add `@xmpp/plugins^0.3.0` dependency. This is the same version bring in by `xmpp.js`
* Add and explicit dependency on `"request": "^2.88.2"` (same version used by `./package.json`).
* Use the full commit hash for `testarmada-magellan-mocha-plugin` dependency

#### Testing instructions

* Verify e2e tests still work
* Run `npx @yarnpkg/doctor` inside `./test/e2e`. All warnings about missing dependencies should be gone now.
